rust_snuba/src/strategies/commit_log.rs

```diff
@@ -81,6 +92,105 @@ impl TryFrom<Commit> for KafkaPayload {
     }
 }
 
+struct ProduceMessage {
+    producer: Arc<dyn Producer<KafkaPayload>>,
+    topic: Arc<TopicOrPartition>,
```

Comment:
Why do you need an Arc for `TopicOrPartition`? If I remember correctly `TopicOrPartition` is an immutable data structure with static lifetime. 
It should be possible to jsut borrow it everywhere.

Reply:
Updated

Reply:
I had to revert this, it is not static currently and would require a large change.

Reply:
Mh do you remember why it is not static ?

Reply:
I am not sure, it's been that way since the beginning. We can look into changing it but I tried and it doesn't seem like an easy change so i'm not going to bundle it in here.

Reply:
Ah now I remember. 
The consumer knows the partitions only after rebalancing, up until the revocation. 
That makes the partitions known to the consumer non static.